For lessons 04 write recipe and 05 write diagnostics (#1)
ACCESS-NRI/CMIP7_MED_Hackathon: _episodes/05-writingdiagnostics.md

## Understanding an existing Python diagnostic

> If you ran `check_hackathon`, a folder called ``ESMValTool`` is
> created in your working directory, this is a clone of the ESMValTool repository.

Reviewer: ```suggestion
A clone of the ESMValTool repository should be available in your user folder in the `nf33`
`scratch` folder (`/scratch/nf33/$USER/ESMValTool`). If not, please make sure to run the
`check_hackathon` command after loading the `esmvaltool` module.
```

- *~/ESMValTool/esmvaltool/recipes/examples/recipe_python.yml*

> - *~/ESMValTool/esmvaltool/diag_scripts/examples/diagnostic.py*

Reviewer: ```suggestion
```
ESMValTool/esmvaltool/recipes/examples/recipe_python.yml
ESMValTool/esmvaltool/diag_scripts/examples/diagnostic.py
```
```

>
{: .callout}

> ## Create a copy of the files for you to edit (Optional)
>
> Copy the files `diagnostic.py` and `recipe_python.yml` to your working folder
> to keep the ones in the repo as templates unaltered while you can more easily 
> find the files you are editing. Edit your recipe to point to your copy of 
> `diagnostic.py`. Also, note the location for when you run your recipe.
>
>> ## Solution
>>
>> Example of your working folder: 

> >> - */scratch/.../Exercise3_files/recipe_python.yml*.
> >> - */scratch/.../Exercise3_files/diagnostic.py*

Reviewer: ```suggestion
>```
>/scratch/nf33/$USER/Exercise3_files/recipe_python.yml
>/scratch/nf33/$USER/Exercise3_files/diagnostic.py
>```
```

>>
>> In your `recipe_python.yml`, edit the path to the diagnostic script.
>> ```yaml
>>     script1:

> >>       script: /scratch/.../Exercise3_files/diagnostic.py

Reviewer: ```suggestion
>>       script: /scratch/nf33/$USER/Exercise3_files/diagnostic.py
```

>>        quickplot:
>> ```
>> When running the recipe run to the full path of your recipe:
>> ```bash

> >> esmvaltool run /scratch/.../Exercise3_files/recipe_python.yml

Reviewer: ```suggestion
>> esmvaltool run /scratch/nf33/$USER/Exercise3_files/recipe_python.yml
```

```yaml
     script1:

>        script: <path_to_script>/diagnostic.py

Reviewer: ```suggestion
       script: <path_to_script diagnostic.py>
```

        quickplot:
          plot_type: pcolormesh
          cmap: Reds
```

This way, we can pass arguments such as the type of
plot ``pcolormesh`` and the colormap ``cmap:Reds`` from the recipe to the
``quickplot`` function in the diagnostic.

> ## Passing arguments from the recipe to the diagnostic
>
> Change the type of the plot and its colormap and inspect the output figure.
>
>> ## Solution
>>
>> In the recipe ``recipe_python.yml``, you could change ``plot_type`` and ``cmap``.
>> As an example, we choose ``plot_type: pcolor`` and ``cmap: BuGn``:
>>
>> ```yaml
>>     script1:

> >>       script: <path_to_script>/diagnostic.py

Reviewer: ```suggestion
>>       script: <path_to_script diagnostic.py>
```
